### ISSUE TITLE
Warnings from C compilation

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#2173]
`PYTHON  setup.py install in code-experiments/build/python` gives (sometimes) these warnings:

```
| code-experiments/src/suite_cons_bbob.c:164:5: warning: code will never be e |
| xecuted [-Wunreachable-code]                                                |
|     coco_free_memory(xopt);                                                 |
|     ^~~~~~~~~~~~~~~~                                                        |
| 1 warning generated.                                                        |
```
and

```
| In file included from /Users/hansen/opt/anaconda3/lib/python3.9/site-packag |
| es/numpy/core/include/numpy/ndarraytypes.h:1969:                            |
| /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/numpy/core/include/ |
| numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API,  |
| disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSI |
| ON" [-W#warnings]                                                           |
| #warning "Using deprecated NumPy API, disable it with " \                   |
```

Full output:

```
$ ./do.py test-postprocessing                                                                                                                       
PYTHON  setup.py install in code-postprocessing
PYTHON  test.py /Users/hansen/opt/anaconda3/bin/python in code-postprocessing/cocopp
PYTHON  generate_aRTA_plot.py in code-postprocessing/aRTAplots
(base) 15:56:46 0 129 coco% ./do.py test-postprocessing-all                                                                                                                    [cwd: ~/git/coco]
PYTHON  setup.py install in code-postprocessing
AML     ['code-experiments/src/coco_random.c', 'code-experiments/src/coco_suite.c', 'code-experiments/src/coco_observer.c', 'code-experiments/src/coco_archive.c', 'code-experiments/src/coco_runtime_c.c'] -> code-experiments/build/python/cython/coco.c
EXPAND  code-experiments/build/python/cython/coco.c.in to code-experiments/build/python/cython/coco.c
EXPAND  code-experiments/src/coco.h to code-experiments/build/python/cython/coco.h
COPY    code-experiments/src/bbob2009_testcases.txt -> code-experiments/build/python/bbob2009_testcases.txt
COPY    code-experiments/src/bbob2009_testcases2.txt -> code-experiments/build/python/bbob2009_testcases2.txt
COPY    code-experiments/build/python/README.md -> code-experiments/build/python/README.txt
EXPAND  code-experiments/build/python/setup.py.in to code-experiments/build/python/setup.py
PYTHON  setup.py install in code-experiments/build/python

An exception occurred while trying to install packages.

A common reason for this error is insufficient access rights
to the installation directory. The original exception message
is as follows:

/----------------------------< EXCEPTION MESSAGE >----------------------------\
|                                                                             |
| NOTE: Using precompiled C file to build interface.                          |
| running install                                                             |
| /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/setuptools/command/ |
| install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated |
| . Use build and pip and other standards-based tools.                        |
|   warnings.warn(                                                            |
| /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/setuptools/command/ |
| easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is |
|  deprecated. Use build and pip and other standards-based tools.             |
|   warnings.warn(                                                            |
| running bdist_egg                                                           |
| running egg_info                                                            |
| writing cocoex.egg-info/PKG-INFO                                            |
| writing dependency_links to cocoex.egg-info/dependency_links.txt            |
| writing top-level names to cocoex.egg-info/top_level.txt                    |
| reading manifest file 'cocoex.egg-info/SOURCES.txt'                         |
| reading manifest template 'MANIFEST.in'                                     |
| adding license file 'LICENSE'                                               |
| adding license file 'AUTHORS'                                               |
| writing manifest file 'cocoex.egg-info/SOURCES.txt'                         |
| installing library code to build/bdist.macosx-11.1-arm64/egg                |
| running install_lib                                                         |
| running build_py                                                            |
| running build_ext                                                           |
| building 'cocoex.interface' extension                                       |
| clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv |
|  -O2 -Wall -fPIC -O2 -isystem /Users/hansen/opt/anaconda3/include -arch arm |
| 64 -I/Users/hansen/opt/anaconda3/include -fPIC -O2 -isystem /Users/hansen/o |
| pt/anaconda3/include -arch arm64 -I/Users/hansen/opt/anaconda3/lib/python3. |
| 9/site-packages/numpy/core/include -I/Users/hansen/opt/anaconda3/include/py |
| thon3.9 -c cython/coco.c -o build/temp.macosx-11.1-arm64-cpython-39/cython/ |
| coco.o                                                                      |
| code-experiments/src/suite_cons_bbob.c:164:5: warning: code will never be e |
| xecuted [-Wunreachable-code]                                                |
|     coco_free_memory(xopt);                                                 |
|     ^~~~~~~~~~~~~~~~                                                        |
| 1 warning generated.                                                        |
| clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv |
|  -O2 -Wall -fPIC -O2 -isystem /Users/hansen/opt/anaconda3/include -arch arm |
| 64 -I/Users/hansen/opt/anaconda3/include -fPIC -O2 -isystem /Users/hansen/o |
| pt/anaconda3/include -arch arm64 -I/Users/hansen/opt/anaconda3/lib/python3. |
| 9/site-packages/numpy/core/include -I/Users/hansen/opt/anaconda3/include/py |
| thon3.9 -c cython/interface.c -o build/temp.macosx-11.1-arm64-cpython-39/cy |
| thon/interface.o                                                            |
| In file included from cython/interface.c:746:                               |
| In file included from /Users/hansen/opt/anaconda3/lib/python3.9/site-packag |
| es/numpy/core/include/numpy/arrayobject.h:4:                                |
| In file included from /Users/hansen/opt/anaconda3/lib/python3.9/site-packag |
| es/numpy/core/include/numpy/ndarrayobject.h:12:                             |
| In file included from /Users/hansen/opt/anaconda3/lib/python3.9/site-packag |
| es/numpy/core/include/numpy/ndarraytypes.h:1969:                            |
| /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/numpy/core/include/ |
| numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API,  |
| disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSI |
| ON" [-W#warnings]                                                           |
| #warning "Using deprecated NumPy API, disable it with " \                   |
|  ^                                                                          |
| 1 warning generated.                                                        |
| clang -bundle -undefined dynamic_lookup -Wl,-rpath,/Users/hansen/opt/anacon |
| da3/lib -L/Users/hansen/opt/anaconda3/lib -L/Users/hansen/opt/anaconda3/lib |
|  -Wl,-rpath,/Users/hansen/opt/anaconda3/lib -L/Users/hansen/opt/anaconda3/l |
| ib build/temp.macosx-11.1-arm64-cpython-39/cython/coco.o build/temp.macosx- |
| 11.1-arm64-cpython-39/cython/interface.o -o build/lib.macosx-11.1-arm64-cpy |
| thon-39/cocoex/interface.cpython-39-darwin.so                               |
| creating build/bdist.macosx-11.1-arm64/egg                                  |
| creating build/bdist.macosx-11.1-arm64/egg/cocoex                           |
| copying build/lib.macosx-11.1-arm64-cpython-39/cocoex/interface.cpython-39- |
| darwin.so -> build/bdist.macosx-11.1-arm64/egg/cocoex                       |
| copying build/lib.macosx-11.1-arm64-cpython-39/cocoex/__init__.py -> build/ |
| bdist.macosx-11.1-arm64/egg/cocoex                                          |
| copying build/lib.macosx-11.1-arm64-cpython-39/cocoex/solvers.py -> build/b |
| dist.macosx-11.1-arm64/egg/cocoex                                           |
| copying build/lib.macosx-11.1-arm64-cpython-39/cocoex/exceptions.py -> buil |
| d/bdist.macosx-11.1-arm64/egg/cocoex                                        |
| copying build/lib.macosx-11.1-arm64-cpython-39/cocoex/utilities.py -> build |
| /bdist.macosx-11.1-arm64/egg/cocoex                                         |
| byte-compiling build/bdist.macosx-11.1-arm64/egg/cocoex/__init__.py to __in |
| it__.cpython-39.pyc                                                         |
| byte-compiling build/bdist.macosx-11.1-arm64/egg/cocoex/solvers.py to solve |
| rs.cpython-39.pyc                                                           |
| byte-compiling build/bdist.macosx-11.1-arm64/egg/cocoex/exceptions.py to ex |
| ceptions.cpython-39.pyc                                                     |
| byte-compiling build/bdist.macosx-11.1-arm64/egg/cocoex/utilities.py to uti |
| lities.cpython-39.pyc                                                       |
| creating stub loader for cocoex/interface.cpython-39-darwin.so              |
| byte-compiling build/bdist.macosx-11.1-arm64/egg/cocoex/interface.py to int |
| erface.cpython-39.pyc                                                       |
| creating build/bdist.macosx-11.1-arm64/egg/EGG-INFO                         |
| copying cocoex.egg-info/PKG-INFO -> build/bdist.macosx-11.1-arm64/egg/EGG-I |
| NFO                                                                         |
| copying cocoex.egg-info/SOURCES.txt -> build/bdist.macosx-11.1-arm64/egg/EG |
| G-INFO                                                                      |
| copying cocoex.egg-info/dependency_links.txt -> build/bdist.macosx-11.1-arm |
| 64/egg/EGG-INFO                                                             |
| copying cocoex.egg-info/top_level.txt -> build/bdist.macosx-11.1-arm64/egg/ |
| EGG-INFO                                                                    |
| writing build/bdist.macosx-11.1-arm64/egg/EGG-INFO/native_libs.txt          |
| zip_safe flag not set; analyzing archive contents...                        |
| cocoex.__pycache__.interface.cpython-39: module references __file__         |
| creating 'dist/cocoex-2.6.1.140-py3.9-macosx-11.1-arm64.egg' and adding 'bu |
| ild/bdist.macosx-11.1-arm64/egg' to it                                      |
| removing 'build/bdist.macosx-11.1-arm64/egg' (and everything under it)      |
| Processing cocoex-2.6.1.140-py3.9-macosx-11.1-arm64.egg                     |
| creating /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/cocoex-2.6 |
| .1.140-py3.9-macosx-11.1-arm64.egg                                          |
| Extracting cocoex-2.6.1.140-py3.9-macosx-11.1-arm64.egg to /Users/hansen/op |
| t/anaconda3/lib/python3.9/site-packages                                     |
| Removing cocoex 2.6.1.139 from easy-install.pth file                        |
| Adding cocoex 2.6.1.140 to easy-install.pth file                            |
|                                                                             |
| Installed /Users/hansen/opt/anaconda3/lib/python3.9/site-packages/cocoex-2. |
| 6.1.140-py3.9-macosx-11.1-arm64.egg                                         |
| Processing dependencies for cocoex==2.6.1.140                               |
| Searching for cocoex==2.6.1.140                                             |
| Reading https://pypi.org/simple/cocoex/                                     |
| Couldn't find index page for 'cocoex' (maybe misspelled?)                   |
| Scanning index of all packages (this may take a while)                      |
| Reading https://pypi.org/simple/                                            |
| No local packages or working download links found for cocoex==2.6.1.140     |
| error: Could not find suitable distribution for Requirement.parse('cocoex== |
| 2.6.1.140')                                                                 |
\-----------------------------------------------------------------------------/
```

**Comment by olafmersmann** [comment]
The `numpy` warnings are because of Cythons use of the deprecated API (see [this](https://github.com/cython/cython/issues/2498) issue for details). Apparently it will be fixed in the Cython 3.0 release.

**Comment by olafmersmann** [comment]
First warning is fixed.
